File: algorithms/66.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution
{
public:
    vector<int> plusOne(vector<int>& digits)
    {
        vector<int> ans;
        int carry = 1;
        for ( int i = digits.size()-1; i >= 0; i-- )
        {
            int sum = digits[i]+carry;
            ans.push_back(sum%10);
            carry = sum/10;
        }
        if ( carry )
            ans.push_back(carry);
        reverse(ans.begin(), ans.end());
        return ans;
    }
};

int main()
{
    return 0;
}

File: algorithms/python/66.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        carry, idx = 1, len(digits)-1
        while carry and idx >= 0:
            t = digits[idx]
            digits[idx] = (t + carry) % 10
            carry = (t + carry) // 10
            idx -= 1
        if carry:
            digits = [carry] + digits
        return digits
